docs/firewall_opnsense.rst

```diff
@@ -0,0 +1,780 @@
+      You may need to turn on DHCP, else you can manually configure a static
+      IPv4 address of ``192.168.1.x`` with a subnet mask of ``255.255.255.0``.
+      However, make sure not to configure your Tails device to have the same IP
+      as the firewall (``192.168.1.1``).
+
+#. The firewall uses a self-signed certificate, so you will see a "This
+   Connection Is Untrusted" warning when you connect. This is expected.
+   You can safely continue by clicking **Advanced** and **Accept the Risk and
+   Continue**.
+
+   |OPNSense - Your Connection is Insecure|
+
+#. You should see the login page for the OPNSense GUI. Log in with the
+   default username and passphrase (``root`` / ``opnsense``).
+
+   |OPNSENSE - Default Login|
+
+If this is your first time logging in to the firewall, the setup wizard will be
+displayed. You should not step through it at this point, however, as there are
+other tasks to complete. To exit, click the OPNSense logo in the top left corner
+of the screen.
+
+Enable Two-Factor Authentication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+OPNSense supports two-factor authentication (2FA) via mobile apps such as Google Authenticator
+or FreeOTP. To set it up, first make sure you have a mobile device available with
+your choice of 2FA app.
+
+Next, in the OPNSense Web GUI, navigate to **System > Access > Servers** and
+click **+** to add a new server.
+
+|OPNSense - auth server|
+
+On the next page, enter ``TOTP Local`` in the **Descriptive name** field and choose
+``Local + Timebased One Time Password`` from the **Type** dropdown. Leave the other
+fields at their default values and click **Save**
+
+Next, navigate to **System > Access > Users** and click the edit button for the ``root``
+user. On the subsequent page, first set a strong admin password. We recomend generating
+a strong passphrase with KeePassXC and saving it in the Tails Persistent folder using
+the provided KeePassXC database template. Then scroll down the page to the **OTP seed**
+section and check the **Generate new secret (160bit)** checkbox. Finally, click **Save**.
+
+|OPNSense - otpcheck|
+
+Once the page has reloaded, scroll down to the **OTP QR code** section and click
+**Click to unhide**, then scan the generated QR code with your mobile auth application
+of choice.
+
+|OPNSense - qrscan|
+
+If you wish, you may also save the OTP seed value displayed above the QR code in
+your Tails KeePassXC database - this isn't required, but will allow you to set up TOTP
+on another mobile device if you need to in the future.
+
+
+Test your new login credentials
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To verify that your new password and OTP secret are working, navigate to **System >
+Access > Tester**. Select ``TOTP Local`` from the **Authentication Server** dropdown,
+enter the ``root`` username in the the **Username** field, and enter your OTP token and password concatenated like ``123456PASSWORD`` in the **Password** field. Then click **Test**.
```

Comment:
`the the`->`the`

Reply:
This is actually a reference to the 80s post-punk band of the same name.

(Ok it isn't, fixed)